File: testdata/bench/swift/BenchmarkTest00029.swift
import Vapor

func handler029(_ req: Request) throws -> Response {
    let firstName = try req.parameters.require("first")
    let lastName = try req.parameters.require("last")
    let html = "<table><tr><td>\(firstName)</td><td>\(lastName)</td></tr></table>"
    return Response(status: .ok, body: .init(string: html))
}

File: testdata/bench/kotlin/BenchmarkTest00029.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*
import io.ktor.http.*

// Vulnerable: XSS via receive body in HTML table
suspend fun handler00029(call: ApplicationCall) {
    val body = call.receive<Map<String, String>>()
    val title = body["title"]
    call.respondText("<table><tr><td>$title</td></tr></table>", ContentType.Text.Html)
}
